core/templates/dev/head/services/speech-synthesis-chunker.service.spec.ts

```diff
@@ -32,87 +31,199 @@ describe('Speech Synthesis Chunker Service', () => {
+        var speakableLatex3 =
+          speechSynthesisChunkerService.formatLatexToSpeakableText(latex3);
+        var speakableLatex4 =
+          speechSynthesisChunkerService.formatLatexToSpeakableText(latex4);
+        expect(speakableLatex1).toEqual('x^2');
+        expect(speakableLatex2).toEqual('42 to the power of 4');
+        expect(speakableLatex3).toEqual('x to the power of 62');
+        expect(speakableLatex4).toEqual('3n to the power of 4x');
+      }
+    );
+
+    it('should properly convert trigonometric functions in LaTeX to ' +
+        'speakable text', () => {
+      var latex1 = '\\\\sin{90}';
+      var latex2 = '\\\\cos{0}';
+      var latex3 = '\\\\tan{uv}';
       var speakableLatex1 =
-        speechSynthesisChunkerService.formatLatexToSpeakableText(latex1);
+          speechSynthesisChunkerService.formatLatexToSpeakableText(latex1);
       var speakableLatex2 =
-        speechSynthesisChunkerService.formatLatexToSpeakableText(latex2);
+          speechSynthesisChunkerService.formatLatexToSpeakableText(latex2);
       var speakableLatex3 =
-        speechSynthesisChunkerService.formatLatexToSpeakableText(latex3);
-      var speakableLatex4 =
-        speechSynthesisChunkerService.formatLatexToSpeakableText(latex4);
-      expect(speakableLatex1).toEqual('x^2');
-      expect(speakableLatex2).toEqual('42 to the power of 4');
-      expect(speakableLatex3).toEqual('x to the power of 62');
-      expect(speakableLatex4).toEqual('3n to the power of 4x');
-    }
-  );
-
-  it ('Should properly convert trigonometric functions in LaTeX to ' +
-      'speakable text', () => {
-    var latex1 = '\\\\sin{90}';
-    var latex2 = '\\\\cos{0}';
-    var latex3 = '\\\\tan{uv}';
-    var speakableLatex1 =
-        speechSynthesisChunkerService.formatLatexToSpeakableText(latex1);
-    var speakableLatex2 =
-        speechSynthesisChunkerService.formatLatexToSpeakableText(latex2);
-    var speakableLatex3 =
-        speechSynthesisChunkerService.formatLatexToSpeakableText(latex3);
-    expect(speakableLatex1).toEqual('the sine of 90');
-    expect(speakableLatex2).toEqual('the cosine of 0');
-    expect(speakableLatex3).toEqual('the tangent of u v');
+          speechSynthesisChunkerService.formatLatexToSpeakableText(latex3);
+      expect(speakableLatex1).toEqual('the sine of 90');
+      expect(speakableLatex2).toEqual('the cosine of 0');
+      expect(speakableLatex3).toEqual('the tangent of u v');
+    });
+  });
+
+  describe('convertToSpeakableText', () => {
+    it('should properly convert to speakable text when raw_latex-with-value' +
+      ' attribute is setted', () => {
+      const html = (
+        '<oppia-noninteractive-math raw_latex-with-value="5-1">' +
+        '</oppia-noninteractive-math>' +
+        '<li>Speech</li>' +
+        '<li>Text</li>'
+      );
+
+      expect(speechSynthesisChunkerService.convertToSpeakableText(html))
+        .toBe('5 minus 1Speech. Text. ');
```

Comment:
are we expecting to have a space between 1 and speech?

Reply:
I've fixed it but I'm not sure if it's the best approach, PTAL